proposals/extensions.md

```diff
@@ -399,54 +399,194 @@ These requirements need more refinement as implementation progresses, and may ne
 
 ### Metadata for declarations
```

Comment:
Can we explain the motivation for this in this doc? If I understood correctly from https://github.com/dotnet/csharplang/blob/main/meetings/working-groups/extensions/content-based-naming.md#summary, it's for EnC, Public API tracking, and cref.